packages/wonder-blocks-modal/util/maybe-get-portal-mounted-modal-host-element.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+import {ModalLauncherPortalAttributeName} from "./constants.js";
+
+/**
+ * From a given element, finds its next ancestor that is a modal launcher portal
+ * element.
+ * @param {null | Element | Text} element The element whose ancestors are to be
+ * walked.
+ * @returns {Element | null} The nearest parent modal launcher portal or
+ * `null`.
+ */
+function maybeGetNextAncestorModalLauncherPortal(element: null | Element | Text) {
+    let candidateElement = element ? element.parentElement : null;
+    while (
+        candidateElement &&
+        !candidateElement.hasAttribute(ModalLauncherPortalAttributeName)
+    ) {
+        candidateElement = candidateElement.parentElement;
+    }
+    return candidateElement || null;
+}
+
+/**
+ * From a given element, finds the next modal host that has been mounted in
+ * a modal portal.
+ * @param {null | Element | Text} element The element whose ancestors are to be
+ * walked.
+ * @returns {Element | null} The next portal-mounted modal host element or
+ * `null`.
+ */
+export default function maybeGetPortalMountedModalHostElement(
+    element: null | Element | Text,
+) {
+    const modalLauncherPortal = maybeGetNextAncestorModalLauncherPortal(element);
+
+    if (modalLauncherPortal) {
+        // If we have a portal, then its first child will be the actual
+        // data-react-root that contains the modal, which is what we want to
+        // return.
+        return modalLauncherPortal.firstElementChild;
```

Comment:
Is that actually true?  Is it not possible for it be the nth child?  Shouldn't `maybeGetNextAncestorModalLauncherPortal` already be return the element that with the `data-modal-launcher-portal` attribute set?

Reply:
`maybeGetNextAncestorModalLauncherPortal` gets the portal with that attribute set, but what we want is the node that is rendered into that portal. Hence this call to get the child. My testing showed that mounting as the last item in the portal itself still had the tooltip render below the modal and its backdrop, whereas rendering within the mounted data-react-root fixed that.

It is plausible that when fragments come with React 16, we might have more than one child, so I could have this always get the last element instead.

Reply:
I wasn't looking at the tests carefully, I see now that we get the parent when we do the check for `data-modal-launcher-portal`.  I'm surprised that we have to render the tooltip into the first child instead of portal itself.

Reply:
Yeah, if I don't then the tooltip appears below the modal and its backdrop.

Reply:
I suspect it has a lot to do with stacking context